cmd/controller/main.go

```diff
@@ -134,9 +134,9 @@ func main() {
 			brokerInformer,
 		),
 	}
-	if len(controllers) != numControllers {
-		logger.Fatalf("Number of controllers and QPS settings mismatch: %d != %d", len(controllers), numControllers)
-	}
+	controllers := controllersArray[:]
```

Comment:
Since this seems to only be used in the `StartAll` call below, we could inline this declaration there:

```go
go kncontroller.StartAll(stopCh, controllersArray[:]...)
```